grunttasks/l10n-htmllint.js

```diff
@@ -0,0 +1,48 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+module.exports = function (grunt) {
+  grunt.registerTask('l10n-htmllint', 'Lint l10n files', [
+    // grunt addition for htmllint
+    'clean',
+    'selectconfig:dist',
+    'l10n-generate-pages',
+    'l10n-json-to-html',
+    'htmllint:l10n'
+    ]
+  );
+
+  grunt.registerMultiTask('l10n-json-to-html', 'Convert l10n JSON files to HTML', function () {
+    var done = this.async();
+    this.files.forEach(function (file) {
+      var content = '';
+      var src = file.src[0];
+      var path = src.split('/');
+      var locale = grunt.file.readJSON(src);
+      for (var val in locale) {
+        var value = locale[val];
+        if (typeof value !== 'object' && value !== '' && isNaN(value)) {
+          content += value.toString() + '\n\n';
+        }
+      }
+      if (content !== '') {
+        grunt.file.write('.tmp/' + path[1] + '/' + path[2] + '/' + path[3].replace('json', 'html'), content);
```

Comment:
need to use `path.join`, see  https://nodejs.org/api/path.html#path_path_join_path1_path2